### PR TITLE
common/TrackedOp: rename and raise prio of slow op perfcounter

### PR DESCRIPTION
**Pull request by YiteGu**
1. TrackedOp is a common class, mon and mds alse use it.
2. Set prio level of l_trackedop_slow_op_count to PRIO_USEFUL, we can get change of slow op of daemon from prometheus metrics, it's very interesting

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by YiteGu** [comment]
```
# ceph -c ceph.conf daemon asok/osd.0.asok perf dump | grep -A 5 trackedop
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
    "trackedop": {
        "slow_ops_count": 0
    }
}
# ceph -c ceph.conf daemon asok/mon.a.asok perf dump | grep -A 5 trackedop
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
    "trackedop": {
        "slow_ops_count": 0
    }
}
# ./ceph -c ceph.conf daemon asok/mds.c.asok perf dump | grep -A 5 trackedop
*** DEVELOPER MODE: setting PATH, PYTHONPATH and LD_LIBRARY_PATH ***
    "trackedop": {
        "slow_ops_count": 0
    }
}
```

**Comment by YiteGu** [comment]
from metrics:
```
# HELP ceph_trackedop_slow_ops_count Number of operations taking over ten second
# TYPE ceph_trackedop_slow_ops_count counter
ceph_trackedop_slow_ops_count{ceph_daemon="mds.c"} 0.0
ceph_trackedop_slow_ops_count{ceph_daemon="mon.a"} 0.0
ceph_trackedop_slow_ops_count{ceph_daemon="mon.b"} 0.0
ceph_trackedop_slow_ops_count{ceph_daemon="mon.c"} 0.0
ceph_trackedop_slow_ops_count{ceph_daemon="osd.0"} 0.0
ceph_trackedop_slow_ops_count{ceph_daemon="osd.1"} 0.0
ceph_trackedop_slow_ops_count{ceph_daemon="osd.2"} 0.0
```

**Comment by YiteGu** [comment]
@rzarzynski @athanatos have a code review, urgent

**Comment by athanatos** [comment]
So, the main change here is to set the priority for l_osd_slow_op_count to PRIO_USEFUL.  Seems reasonable to me. @rzarzynski @batrick Does this seem reasonable to you?

**Review by athanatos — APPROVED** [review]

**Review by batrick — CHANGES_REQUESTED** [review]
Please rewrite commit title as: `common/TrackedOp: rename and raise prio of slow op perfcounter`

**Comment by batrick** [comment]
> So, the main change here is to set the priority for l_osd_slow_op_count to PRIO_USEFUL. Seems reasonable to me. @rzarzynski @batrick Does this seem reasonable to you?

Yup!

File: src/common/TrackedOp.h [hunk 1]
```diff
@@ -75,9 +75,11 @@ class OpHistory {
 
 public:
   OpHistory(CephContext *c) : cct(c), opsvc(this) {
-    PerfCountersBuilder b(cct, "osd-slow-ops",
-                         l_osd_slow_op_first, l_osd_slow_op_last);
-    b.add_u64_counter(l_osd_slow_op_count, "slow_ops_count",
+    PerfCountersBuilder b(cct, "trackedop",
+                         l_trackedop_slow_op_first, l_trackedop_slow_op_last);
+    b.set_prio_default(PerfCountersBuilder::PRIO_USEFUL);
```
Comment (rzarzynski) [inline src/common/TrackedOp.h:80]: Yeah, I think there shouldn't be too much of worries about the renaming as this counter was `PRIO_DEBUGONLY` only.

```cpp
class PerfCountersBuilder
{
public:
  PerfCountersBuilder(CephContext *cct, const std::string &name,
                    int first, int last);
...
  // prio values: higher is better, and higher values get included in
  // 'ceph daemonperf' (and similar) results.
  // Use of priorities enables us to add large numbers of counters
  // internally without necessarily overwhelming consumers.
  enum {
    PRIO_CRITICAL = 10,
    // 'interesting' is the default threshold for `daemonperf` output
    PRIO_INTERESTING = 8,
    // `useful` is the default threshold for transmission to ceph-mgr
    // and inclusion in prometheus/influxdb plugin output
    PRIO_USEFUL = 5,
    PRIO_UNINTERESTING = 2,
    PRIO_DEBUGONLY = 0,
  };
...
  void set_prio_default(int prio_)
  {
    prio_default = prio_;
  }
...
  int prio_default = 0;
};

**Review by rzarzynski — APPROVED** [review]
LGTM assuming the Patrick's review gets addressed.

**Comment by YiteGu** [comment]
@batrick Rewrite commit title done

**Review by batrick — APPROVED** [review]

**Comment by YiteGu** [comment]
results show
截屏2023-08-01 下午1 56 15

**Comment by YiteGu** [comment]
Hi @ljflores this pr need testing

**Comment by YiteGu** [comment]
@rzarzynski I think we need to fix this problem, otherwise, the slow op of mon and mds will be represented by the keyword osd, this will cause ambiguity

**Comment by batrick** [comment]
ping @ljflores @rzarzynski @yuriw please QA.

**Comment by rzarzynski** [comment]
jenkins test make check

**Comment by rzarzynski** [comment]
`needs-qa` is already assigned.

@yuriw, @ljflores: let's take this into the very next QA batch. Pinging also in Slack.

**Comment by rzarzynski** [comment]
Looks unrelated at first glance:

```
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/bin/nodeenv", line 8, in 
    sys.exit(main())
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/lib/python3.10/site-packages/nodeenv.py", line 1130, in main
    create_environment(env_dir, args)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/lib/python3.10/site-packages/nodeenv.py", line 1006, in create_environment
    install_node(env_dir, src_dir, args)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/lib/python3.10/site-packages/nodeenv.py", line 763, in install_node
    install_node_wrapped(env_dir, src_dir, args)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/lib/python3.10/site-packages/nodeenv.py", line 786, in install_node_wrapped
    download_node_src(node_url, src_dir, args)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/lib/python3.10/site-packages/nodeenv.py", line 618, in download_node_src
    dl_contents = _download_node_file(node_url)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/lib/python3.10/site-packages/nodeenv.py", line 610, in _download_node_file
    raise e
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/src/pybind/mgr/dashboard/frontend/node-env/lib/python3.10/site-packages/nodeenv.py", line 602, in _download_node_file
    return io.BytesIO(urlopen(node_url).read())
  File "/usr/lib/python3.10/http/client.py", line 482, in read
    s = self._safe_read(self.length)
  File "/usr/lib/python3.10/http/client.py", line 633, in _safe_read
    raise IncompleteRead(data, amt-len(data))
http.client.IncompleteRead: IncompleteRead(12623872 bytes read, 34097368 more expected)
```

**Comment by rzarzynski** [comment]
jenkins test make check

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrackercephcomissues67090

**Pull request merged**